### PR TITLE
[BD-24] [BB-3245] LTI 1.3 Studio preview window

### PR DESCRIPTION
**Pull request by shimulch**
This PR adds a new ``Test Launch`` link on LTI author view in Studio. Clicking on that link performs an LTI launch in a new window.

**JIRA tickets**:  https://tasks.opencraft.com/browse/BB-3245

~~**Discussions**: ~~

**Dependencies**: None

**Screenshots**: 

lti_preview

~~**Sandbox URL**:~~

**Merge deadline**: None

**Testing instructions**:

1. Install [xblock-lti-consumer](https://github.com/edx/xblock-lti-consumer) on local devstack.
2. Make sure LTI 1.3 enabled by setting the feature flag on ``studio.yml``
```
FEATURES:
    LTI_1P3_ENABLED: true
```
3. Create an LTI 1.3 block. Follow the procedure from [here](https://github.com/edx/xblock-lti-consumer#lti-13).
4. Verify that ``Test Launch`` link visible.
5. After publishing, click the ``Test Launch`` link. It should open a new window and perform an LTI launch.

**Author notes and concerns**:

1. I've tried to do it within an iframe. But the launch couldn't be performed as LMS is not of the same origin, thus not loaded inside iframe during auth flow. So I found it easier to open it on a new window instead.

**Reviewers**
- [ ] @viadanna 
- [ ] edX reviewer[s] TBD

~~**Settings**~~

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @shimulch! I've created [BLENDED-670](https://openedx.atlassian.net/browse/BLENDED-670) to keep track of it in Jira.   More details are on the [BD-24](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1545076784/BD-24+LTI+v1.3+Improvements+Advantage+Support) project page.


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

File: lti_consumer/lti_xblock.py
Comment (viadanna) [inline lti_consumer/lti_xblock.py:900]: Didn't expect a URL ending in `?`. Is it necessary?

**Review by viadanna — APPROVED** [review]
Good to go 👍

Just a minor nit/question.

- [x] I tested this on a master devstack.
- [x] I read through the code.

File: lti_consumer/templates/html/lti_1p3_studio.html
Comment (viadanna) [inline lti_consumer/templates/html/lti_1p3_studio.html:51]: Can you make sure the button displays as the mockup?

image

**Review by viadanna — CHANGES_REQUESTED** [review]
Sorry, forgot about the button style. Not that I expect a lot of work here for that ;)

File: lti_consumer/lti_xblock.py
Comment (shimulch) [reply lti_consumer/lti_xblock.py:900]: @viadanna I've copied it from [here](https://github.com/edx/xblock-lti-consumer/blob/9ac5fda45c352fc456ab8b432bb967faf1a4661a/lti_consumer/lti_xblock.py#L1363). Not sure why that's there. I thought it might be there for some kind of issue before. Should I remove it?

Comment (viadanna) [reply lti_consumer/lti_xblock.py:900]: In that case, let's keep it, I might be missing some context here.

File: lti_consumer/templates/html/lti_1p3_studio.html
Comment (shimulch) [reply lti_consumer/templates/html/lti_1p3_studio.html]: @viadanna, Used a button instead of a link now. The style should be ok.

**Review by viadanna — APPROVED** [review]
Good to go 👍

- [x] I tested this on a sandbox.
- [x] I read through the code.

image

**Comment by shimulch** [comment]
@viadanna, Thanks for approving this.

@nedbat, This PR is ready for review.

**Comment by giovannicimolin** [comment]
@shimulch This is missing a version bump.

**Comment by shimulch** [comment]
@giovannicimolin, When there are multiple PRs, how do we actually manage version changes? Is it a good idea to wait till getting PR approved by edX and then add version change based on the current version on master at that time? :)

**Comment by giovannicimolin** [comment]
@shimulch Yup, that approach makes sense since version bumps are always changing. You can wait for approval before merging pushing the version bump.

**Pull request closed without merging**
